plugins/llvm/llvm_binary.hpp

```diff
@@ -434,39 +345,67 @@ using namespace llvm;
 using namespace llvm::object;
 
 struct section {
-    explicit section(const SectionRef& sec) {
-        StringRef name;
-        if(error_code err = sec.getName(name))
-            llvm_binary_fail(err);
-        this->name_ = name.str();
-        if (error_code err = sec.getAddress(this->addr_))
-            llvm_binary_fail(err);
-
-        if (error_code err = sec.getSize(this->size_))
-            llvm_binary_fail(err);
-    }
-    const std::string& name() const { return name_; }
-    uint64_t addr() const { return addr_; }
-    uint64_t size() const { return size_; }
-
-private:
-    std::string name_;
-    uint64_t addr_;
-    uint64_t size_;
+    std::string name;
+    uint64_t addr;
+    uint64_t size;
 };
 
-std::vector<section> read(const ObjectFile& obj) {
+section make_section(const SectionRef &sec) {
+    StringRef name;
+    if (error_code err = sec.getName(name))
+	llvm_binary_fail(err);
+    
+    uint64_t addr;
+    if (error_code err = sec.getAddress(addr))
+	llvm_binary_fail(err);
+
+    uint64_t size;
+    if (error_code err = sec.getSize(size))
+	llvm_binary_fail(err);
+    return section{name.str(), addr, size};
+}
+
+std::vector<section> read(const ObjectFile &obj) {
     int size = utils::distance(obj.begin_sections(),
                                obj.end_sections());
     std::vector<section> sections;
     sections.reserve(size);
     std::transform(obj.begin_sections(),
                    obj.end_sections(),
                    std::back_inserter(sections),
-                   [](const SectionRef& s) { return section(s); });
+                   [](const SectionRef& s) { return make_section(s); });
     return sections;
 }
 
+section make_section(const coff_section &s, const uint64_t image_base) {
+    return section{s.Name, s.VirtualAddress + image_base, s.SizeOfRawData};
+}
+
+template <typename T>
+std::vector<section> readPE(const COFFObjectFile &obj, const T image_base) {
+    std::vector<section> sections;
+    for (auto it = obj.begin_sections();
+         it != obj.end_sections(); ++it) {
+        const coff_section *s = obj.getCOFFSection(it);
```

Comment:
shouldn't we check that the `coff_section` is not `null`? In the rest part of the code we do the check...